### PR TITLE
WL-4539 - Elfinder will not load if user is a member of an AV site

### PR DESCRIPTION
**Pull request by mpellicer**
Elfinder will not load if user is a member of an AV site

### CODE REVIEW

**Comment by buckett** [comment]
I think we shouldn't do a two factor lookup but should instead just cope with a site that you are a member of that you can't see at yet. Is that possible?

**Comment by mpellicer** [comment]
First at all i thought to use the siteService.allowAccessSite() method but going deep i've noticed it doesn't check if the user is authenticated using the secondary method, that's why i've implemented the two factor lookup.

Also i've noticed the site appears in the site list despite the user is not authenticated in the secondary entity, is it correct?

Let me dig a bit more and make another implementation.

Thanks Matt.

**Comment by mpellicer** [comment]
@buckett  

I've done a new implementation, the unlock method from SecurityService also checks the TwoFactorAuthentication.

File: textarea/elfinder-sakai/src/java/org/sakaiproject/elfinder/sakai/SakaiFsService.java [hunk 1]
```diff
@@ -179,6 +181,10 @@ public String getVolumeId(FsVolume volume) {
 			if("!admin".equals(currentSiteId) || "~admin".equals(currentSiteId)){
 				continue;
 			}
+			//Exclude a site where the user does not have access because it's secure.
+			if(!securityService.unlock(siteService.SITE_VISIT, "/site/"+currentSiteId)){
```
Comment (buckett) [inline textarea/elfinder-sakai/src/java/org/sakaiproject/elfinder/sakai/SakaiFsService.java:185]: Calling `siteService.allowGetSite(site)` would seem to be the better API to call here.

**Pull request closed without merging**